sklearn/utils/validation.py

```diff
@@ -478,7 +478,7 @@ def check_array(array, accept_sparse=False, accept_large_sparse=True,
     # DataFrame), and store them. If not, store None.
     dtypes_orig = None
     if hasattr(array, "dtypes") and hasattr(array, "__array__"):
-        dtypes_orig = np.array(array.dtypes)
+        dtypes_orig = np.array(array.dtypes, ndmin=1)
```

Comment:
I am not sure this is the "correct" fix. It fixes the problem, but in essence, a Series should not take this path as it can never have multiple dtypes in it. So I would rather ensure that a Series does not pass this `if` check

Reply:
@qinhanmin2014 had one in https://github.com/scikit-learn/scikit-learn/issues/12622#issuecomment-440322835

I am trying to think of a robust "duck type check" for Series .. Personally, I would actually start doing actual isinstance checks (we could eg have a util function that combines that with trying to import pandas), but that's maybe a broader issue.

Reply:
we could also do ``and array.dtypes.ndim`` so it stays ``None``?

Reply:
Yes, that's a good idea (for sure nicer than the `hasattr(array.dtypes, "__array__")`)